packages/firebase_dynamic_links/ios/Classes/FLTFirebaseDynamicLinksPlugin.m

```diff
@@ -118,63 +118,57 @@ - (NSMutableDictionary *)getInitialLink {
   return getDictionaryFromDynamicLink(_initialLink);
 }
 
-- (BOOL)application:(UIApplication *)application
-            openURL:(NSURL *)url
-            options:(NSDictionary<UIApplicationOpenURLOptionsKey, id> *)options {
-  return [self checkForDynamicLink:url];
-}
-
-- (BOOL)application:(UIApplication *)application
-              openURL:(NSURL *)url
-    sourceApplication:(NSString *)sourceApplication
-           annotation:(id)annotation {
-  return [self checkForDynamicLink:url];
+- (void) onDeepLinkResult: (FIRDynamicLink *_Nullable)dynamicLink
+                    error: (NSError *_Nullable) error {
+    if (_initiated) {
+        if (error) {
+            FlutterError *flutterError = getFlutterError(error);
+            [_channel invokeMethod:@"onLinkError"
+                         arguments:getDictionaryFromFlutterError(flutterError)];
+        } else {
+            NSMutableDictionary *dictionary = getDictionaryFromDynamicLink(dynamicLink);
+            [_channel invokeMethod:@"onLinkSuccess" arguments:dictionary];
+        }
+    } else {
+        if (error) {
+            _flutterError = getFlutterError(error);
+        } else {
+            _initialLink = dynamicLink;
+        }
+    }
 }
 
 - (BOOL)checkForDynamicLink:(NSURL *)url {
   FIRDynamicLink *dynamicLink = [[FIRDynamicLinks dynamicLinks] dynamicLinkFromCustomSchemeURL:url];
   if (dynamicLink) {
-    if (dynamicLink.url) _initialLink = dynamicLink;
+    [self onDeepLinkResult: dynamicLink error: nil];
```

Comment:
info: when the `checkForDynamicLink` method is executed, the `getInitialLink()` method had been already called. In such a case the app won't read `_initialLink`.
After the change, the link is delivered to `onLink(onSuccess: xyz)` callback